### PR TITLE
Make tools/bcp47.py compatible with python3

### PR DESCRIPTION
**Pull request by stephen-huan**
python2 has been [depreciated](https://www.python.org/doc/sunset-python-2/) since 2020. It'd be useful if I could use this file for some translation work in pandoc, and I'm writing code in python3. The first commit makes the script usable from python3, while keeping compatibility with python2 (>= 2.6). The next two commits are some minor nits.

The third commit fixes a bug, currently, `babelname2bcp47` maps `chinese` to `zh-TW` since a dictionary with duplicate keys takes the _last_ value. [Babel](https://github.com/latex3/babel/blob/dcd551427599879303b7a047b6fd3aef0a77120e/locale/zh/babel-zh.ini#L19-L20) uses `chinese` to mean `zh`, so the duplicates have been removed. The key `sa-Deva` is also duplicated in `bcp472lang`, but the value is the same for both so the resulting semantics are the same.

The fourth commit adds some missing bcp47 codes to `bcp472lang`.

### CODE REVIEW

**Comment by Udi-Fogiel** [comment]
Thanks! I'm actually planing to port this script to lua, as I would like to have less dependencies. Will that be a problem for your projects? 

Note that I'm less familiar with this script, as I did not wrote it nor used it before, so unless @jspitz beats me to it  it will probably take me a couple of days to find a time to review this PR.

**Comment by stephen-huan** [comment]
> Thanks! I'm actually planing to port this script to lua, as I would like to have less dependencies. Will that be a problem for your projects?

In the short term, I base my work off my fork of polyglossia, so it wouldn't directly affect me.  In the long run, I've been thinking about moving to lua myself (for the same reason, pandoc embeds a lua interpreter by default). So in either case, you shouldn't need to worry about my (relatively niche) use case.

> Note that I'm less familiar with this script, as I did not wrote it nor used it before, so unless @jspitz beats me to it it will probably take me a couple of days to find a time to review this PR.

No worries, I made the modifications and thought I might as well upstream the fixes.

File: tools/bcp47.py [hunk 1]
```diff
@@ -32,8 +35,6 @@
     "canadien" : "fr-CA",
     "catalan" : "ca",
     "chinese" : "zh",
-    "chinese" : "zh-CN",
-    "chinese" : "zh-TW",
```
Comment (jspitz) [inline tools/bcp47.py:36]: Why these deletions?

File: tools/bcp47.py [hunk 2]
```diff
@@ -262,7 +265,6 @@
     "sa" : "sanskrit",
     "sa-Deva" : "sanskrit",
     "sa-Beng" : "sanskrit",
-    "sa-Deva" : "sanskrit",
```
Comment (jspitz) [inline tools/bcp47.py:265]: dito

File: tools/bcp47.py (hunk 1 above)
Comment (stephen-huan) [reply tools/bcp47.py:36]: Python dictionaries can't have the same key map to multiple values. In the case of a conflict, the _last_ value defined is taken. For example,
```python
>>> {"chinese": "zh", "chinese": "zh-CN", "chinese": "zh-TW"}
{'chinese': 'zh-TW'}
```
[Babel](https://github.com/latex3/babel/blob/dcd551427599879303b7a047b6fd3aef0a77120e/locale/zh/babel-zh.ini#L19-L20) uses `chinese` to mean `zh`, so the other two have been removed.

File: tools/bcp47.py (hunk 2 above)
Comment (stephen-huan) [reply tools/bcp47.py:265]: The key `sa-Deva` is already defined with the same value two lines above the deleted line, so this line can be safely removed. See the response to the other comment as well (https://github.com/reutenauer/polyglossia/pull/649#discussion_r1675718376).

Comment (jspitz) [reply tools/bcp47.py:265]: thanks

**Pull request merged**